lib/reader/src/lexer.rs

```diff
@@ -326,6 +338,39 @@ impl<'a> Lexer<'a> {
         }
     }
 
+    fn scan_name(&mut self) -> Result<LocatedToken<'a>, LocatedError> {
+        let loc = self.loc();
+        let begin = self.pos;
+
+        assert!(self.lookahead == Some('%'));
+
+        while let Some(c) = self.next_ch() {
+            if !c.is_alphanumeric() && c != '_' {
```

Comment:
I can't seem to find any stdlib functions to check for ascii-only alphabetic characters, you don't know if there are any do you?